ffi/passmanagers.cpp

```diff
@@ -24,164 +41,380 @@ using namespace llvm;
+}
+
+API_EXPORT(void)
+LLVMPY_AddAggressiveInstructionCombiningPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createAggressiveInstCombinerPass());
+}
+
+API_EXPORT(void)
+LLVMPY_AddInternalizePass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createInternalizePass());
+}
+
+API_EXPORT(void)
+LLVMPY_AddJumpThreadingPass(LLVMPassManagerRef PM, int threshold) {
+  unwrap(PM)->add(createJumpThreadingPass(threshold));
+}
+
+API_EXPORT(void)
+LLVMPY_AddLCSSAPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createLCSSAPass());
+}
+
+API_EXPORT(void)
+LLVMPY_AddLoopDeletionPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createLoopDeletionPass());
+}
+
+API_EXPORT(void)
+LLVMPY_AddLoopExtractorPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createLoopExtractorPass());
+}
+
+API_EXPORT(void)
+LLVMPY_AddSingleLoopExtractorPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createSingleLoopExtractorPass());
+}
+
+API_EXPORT(void)
+LLVMPY_AddLoopStrengthReducePass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createLoopStrengthReducePass());
+}
+
+API_EXPORT(void)
+LLVMPY_AddLoopSimplificationPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createLoopSimplifyPass());
+}
+
+API_EXPORT(void)
+LLVMPY_AddLoopUnrollPass(LLVMPassManagerRef PM) { LLVMAddLoopUnrollPass(PM); }
+
+API_EXPORT(void)
+LLVMPY_AddLoopUnrollAndJamPass(LLVMPassManagerRef PM) {
+  LLVMAddLoopUnrollAndJamPass(PM);
+}
+
+API_EXPORT(void)
+LLVMPY_AddLoopUnswitchPass(LLVMPassManagerRef PM, bool optimizeForSize,
+                           bool hasBranchDivergence) {
+  unwrap(PM)->add(createLoopUnswitchPass(optimizeForSize, hasBranchDivergence));
 }
 
 API_EXPORT(void)
-LLVMPY_AddConstantMergePass(LLVMPassManagerRef PM)
-{
-    LLVMAddConstantMergePass(PM);
+LLVMPY_AddLowerAtomicPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createLowerAtomicPass());
 }
 
 API_EXPORT(void)
-LLVMPY_AddDeadArgEliminationPass(LLVMPassManagerRef PM)
-{
-    LLVMAddDeadArgEliminationPass(PM);
+LLVMPY_AddLowerInvokePass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createLowerInvokePass());
 }
 
 API_EXPORT(void)
-LLVMPY_AddFunctionAttrsPass(LLVMPassManagerRef PM)
-{
-    LLVMAddFunctionAttrsPass(PM);
+LLVMPY_AddLowerSwitchPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createLowerSwitchPass());
 }
 
 API_EXPORT(void)
-LLVMPY_AddFunctionInliningPass(LLVMPassManagerRef PM, int Threshold)
-{
-    unwrap(PM)->add(createFunctionInliningPass(Threshold));
+LLVMPY_AddMemCpyOptimizationPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createMemCpyOptPass());
 }
 
 API_EXPORT(void)
-LLVMPY_AddGlobalOptimizerPass(LLVMPassManagerRef PM)
-{
-    LLVMAddGlobalOptimizerPass(PM);
+LLVMPY_AddMergeFunctionsPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createMergeFunctionsPass());
 }
 
 API_EXPORT(void)
-LLVMPY_AddGlobalDCEPass(LLVMPassManagerRef PM)
-{
-    LLVMAddGlobalDCEPass(PM);
+LLVMPY_AddMergeReturnsPass(LLVMPassManagerRef PM) {
+  unwrap(PM)->add(createUnifyFunctionExitNodesPass());
```

Comment:
LLVMPY and LLVM names don't correlate? Is the intent for the LLVMPY API to match naming closer to the flags for pass names provided to `opt`?

Reply:
The LLVM function names don't match the names in the documentation. I was using the names in the documentation since I assumed that would be the reference used.